tensorly/kruskal_tensor.py

```diff
@@ -36,12 +41,14 @@ def kruskal_to_tensor(factors, weights=None):
 
     There are other possible and equivalent alternate implementation, e.g.
     summing over r and updating an outer product of vectors.
+
     """
     shape = [T.shape(factor)[0] for factor in factors]
     if weights is not None:
-        full_tensor = T.dot(factors[0]*weights, T.transpose(khatri_rao(factors[1:])))
+        factors[0] = factors[0]*weights
+        full_tensor = T.sum(khatri_rao(factors, mask=mask), axis=1)
     else:
-        full_tensor = T.dot(factors[0], T.transpose(khatri_rao(factors[1:])))
+        full_tensor = T.sum(khatri_rao(factors, mask=mask), axis=1)
```

Comment:
This will use more memory and I assume will be a little slower?

Reply:
Sorry it's been a while since I've looked at this. I don't remember why I made this change. If it still works and is sparse safe after reverting it (see https://github.com/JeanKossaifi/tensorly-notebooks/blob/7179c121cc2f21215f9e14cd1b452f8576683c48/07_pydata_sparse_backend/sparse_missing_values.ipynb), that's fine.